lib/readline.js

```diff
@@ -181,7 +181,11 @@ function Interface(input, output, completer, terminal) {
   }
 
   function onkeypress(s, key) {
-    self._ttyWrite(s, key);
+    if (process.env.TERM === 'dumb') {
+      self._ttyWriteDumb(s, key);
+    } else {
+      self._ttyWrite(s, key);
+    }
```

Comment:
I personally would check the env while starting the readline instance and just set `instance._ttyWrite = _ttyWriteDumb.bind(instance)`. That way we have to check less often for the env and in less places.

However, there is no right or wrong so far and it's discussed in https://github.com/nodejs/node/issues/26213 if we want to adhere to a specific way to check for envs.

Reply:
Ok, it will eliminate 2 checks, one in the frequent code path.